spec/services/events/stores/aggregated_clickhouse_store_spec.rb

```diff
@@ -1186,4 +1186,255 @@
+        {timestamp: Time.zone.parse("2023-03-01 04:00:00"), value: -2, agent_name: "aragorn"},
+        {timestamp: Time.zone.parse("2023-03-01 05:00:00"), value: 10, agent_name: "aragorn"},
+        {timestamp: Time.zone.parse("2023-03-01 05:30:00"), value: -10, agent_name: "aragorn"},
+
+        {timestamp: Time.zone.parse("2023-03-01 00:00:00.000"), value: 2},
+        {timestamp: Time.zone.parse("2023-03-01 01:00:00"), value: 3},
+        {timestamp: Time.zone.parse("2023-03-01 01:30:00"), value: 1},
+        {timestamp: Time.zone.parse("2023-03-01 02:00:00"), value: -4},
+        {timestamp: Time.zone.parse("2023-03-01 04:00:00"), value: -2},
+        {timestamp: Time.zone.parse("2023-03-01 05:00:00"), value: 10},
+        {timestamp: Time.zone.parse("2023-03-01 05:30:00"), value: -10}
+      ]
+    end
+
+    let(:events) do
+      events_values.map do |values|
+        ::Clickhouse::EventsEnrichedExpanded.create!(
+          transaction_id: SecureRandom.uuid,
+          organization_id: organization.id,
+          external_subscription_id: subscription.external_id,
+          subscription_id: subscription.id,
+          plan_id: plan.id,
+          code:,
+          aggregation_type: "unique_count",
+          charge_id:,
+          charge_version: charge.updated_at,
+          charge_filter_id: charge_filter&.id,
+          charge_filter_version: charge_filter&.updated_at,
+          timestamp: values[:timestamp],
+          properties: {},
+          value: values[:value].to_s,
+          decimal_value: values[:value].to_d,
+          grouped_by: {
+            "agent_name" => values[:agent_name] || described_class::NIL_GROUP_VALUE,
+            "other" => described_class::NIL_GROUP_VALUE
+          }
+        )
+      end
+    end
+
+    before do
+      event_store.aggregation_property = billable_metric.field_name
+      event_store.numeric_property = true
+    end
+
+    it "returns the weighted sum of event properties" do
+      result = event_store.grouped_weighted_sum
+
+      expect(result.count).to eq(3)
+
+      null_group = result.find { |v| v[:groups]["agent_name"].nil? }
+      expect(null_group[:groups]["agent_name"]).to be_nil
+      expect(null_group[:groups]["other"]).to be_nil
+      expect(null_group[:value].round(5)).to eq(0.02218)
+
+      (result - [null_group]).each do |row|
+        expect(row[:groups]["agent_name"]).not_to be_nil
+        expect(row[:groups]["other"]).to be_nil
+        expect(row[:value].round(5)).to eq(0.02218)
+      end
+    end
+
+    context "with no events" do
+      let(:events_values) { [] }
+
+      it "returns the weighted sum of event properties" do
+        result = event_store.grouped_weighted_sum
+
+        expect(result.count).to eq(0)
+      end
+    end
+
+    context "with initial values" do
+      let(:initial_values) do
+        [
+          {groups: {"agent_name" => "frodo", "other" => nil}, value: 1000},
```

Comment:
It would be nice to use different initial values and different groups values to properly test the results. Here if we return the same result, in the query for all groups, the test would still pass.